Move passing examples to test_data to be automatically tested

jfecher opened the pull request:
Also renames `examples` to `examples_failing` to make the goal of making these all eventually pass more clear.

There are still a few remaining examples in examples_failing that fail to compile, mostly from crashes either in ssa or during verification.

vezenovm reviewed (commented):
As this PR just moves passing examples into `test_data` and the CI is passing it all looks good. What do you think about possibly renaming the remaining tests in `examples_failing` like we did in `test_data` from a number to the specific functionality that is being tested. Makes it a little easier for when searching through the examples for the first time.

jfecher commented:
Renamed the remaining numeric examples. I also removed example 15, it only tested set_as_public which isn't supported by ssa and is being removed from the spec in favor of #232

vezenovm reviewed (approved):
LGTM
